wallet/internal/db/queries/sqlite/transactions.sql

```diff
@@ -0,0 +1,169 @@
+-- name: InsertTransaction :one
+-- Inserts a new transaction record.
+-- For confirmed transactions, block_height must be set.
+-- For unconfirmed transactions, block_height should be NULL.
+-- Returns the auto-generated ID.
+INSERT INTO transactions (tx_hash, block_height, is_coinbase, received_timestamp, serialized_tx, tx_label)
+VALUES (?, ?, ?, ?, ?, ?)
+ON CONFLICT (tx_hash, block_height) DO NOTHING
+RETURNING id;
+
+-- name: GetTransactionByHash :one
+-- Gets a transaction by hash with priority: unconfirmed first, then highest confirmed block.
+SELECT
+    t.id,
+    t.tx_hash,
+    t.block_height,
+    t.is_coinbase,
+    t.received_timestamp,
+    t.serialized_tx,
+    t.tx_label,
+    b.header_hash,
+    b.block_timestamp
+FROM transactions AS t
+LEFT JOIN blocks AS b ON t.block_height = b.block_height
+WHERE t.tx_hash = ?
+ORDER BY t.block_height DESC NULLS FIRST
+LIMIT 1;
+
+-- name: GetTransactionByHashAndBlock :one
+-- Retrieves a confirmed transaction by its hash and block height.
+SELECT
+    id,
+    tx_hash,
+    block_height,
+    is_coinbase,
+    received_timestamp,
+    serialized_tx,
+    tx_label
+FROM transactions
+WHERE tx_hash = ? AND block_height = ?;
+
+-- name: GetUnconfirmedTransactionByHash :one
+-- Retrieves an unconfirmed transaction by its hash.
+SELECT
+    id,
+    tx_hash,
+    block_height,
+    is_coinbase,
+    received_timestamp,
+    serialized_tx,
+    tx_label
+FROM transactions
+WHERE tx_hash = ? AND block_height IS NULL;
+
+-- name: GetLatestConfirmedTransactionByHash :one
+-- Gets the most recent CONFIRMED (mined) transaction with a given hash.
+-- Returns only confirmed transactions, ordered by highest block.
+SELECT
+    id,
+    tx_hash,
+    block_height,
+    is_coinbase,
+    received_timestamp,
+    serialized_tx,
+    tx_label
+FROM transactions
+WHERE tx_hash = ? AND block_height IS NOT NULL
+ORDER BY block_height DESC
+LIMIT 1;
+
+-- name: TransactionExists :one
+-- Checks if a transaction with the given hash and block exists.
+SELECT CASE WHEN EXISTS (
+    SELECT 1 FROM transactions
+    WHERE tx_hash = ? AND block_height = ?
+) THEN 1 ELSE 0 END;
```

Comment:
Why not just like this?

```suggestion
SELECT EXISTS (
    SELECT 1 FROM transactions
    WHERE tx_hash = ? AND block_height = ?
);
```


> If executing the SELECT statement specified as the right-hand operand of the EXISTS operator would return one or more rows, then the EXISTS operator evaluates to 1. If executing the SELECT would return no rows at all, then the EXISTS operator evaluates to 0.